### PR TITLE
fix ingress for k8s v1.19

### LINKED ISSUE #13172: ingress: unknown field "ipFamilies" in io.k8s.api.core.v1.ServiceSpec with k8s v1.19
### What Happened?

as reported by Yevhenii Dytyniuk in slack thread: https://kubernetes.slack.com/archives/C1F5CT6Q1/p1639259539285100?thread_ts=1639259222.284200&cid=C1F5CT6Q1, and acknowledged upstream: https://github.com/kubernetes/ingress-nginx/issues/7810

### Attach the log file

```
stderr:
error: error validating "/etc/kubernetes/addons/ingress-deploy.yaml": error validating data: [ValidationError(Service.spec): unknown field "ipFamilies" in io.k8s.api.core.v1.ServiceSpec, ValidationError(Service.spec): unknown field "ipFamilyPolicy" in io.k8s.api.core.v1.ServiceSpec]; if you choose to ignore these errors, turn validation off with --validate=false
 waiting for app.kubernetes.io/name=ingress-nginx pods: timed out waiting for the condition]
```

### Operating System

_No response_

### Driver

_No response_

### PR DESCRIPTION
**Pull request by prezha**
fixes #13172

bonus: upgrades ingress-nginx/controller to the latest v1.1.0

example output:

```
❯ minikube start --kubernetes-version=v1.19.6
😄  minikube v1.24.0 on Opensuse-Tumbleweed
✨  Automatically selected the docker driver. Other choices: kvm2, ssh
💨  For improved Docker performance, enable the overlay Linux kernel module using 'modprobe overlay'
❗  docker is currently using the btrfs storage driver, consider switching to overlay2 for better performance
👍  Starting control plane node minikube in cluster minikube
🚜  Pulling base image ...
🔥  Creating docker container (CPUs=2, Memory=16000MB) ...
🐳  Preparing Kubernetes v1.19.6 on Docker 20.10.11 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔎  Verifying Kubernetes components...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🌟  Enabled addons: storage-provisioner, default-storageclass

❗  /home/prezha/bin/k8s/kubectl is version 1.23.0, which may have incompatibilites with Kubernetes 1.19.6.
    ▪ Want kubectl v1.19.6? Try 'minikube kubectl -- get pods -A'
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```
```
❯ minikube addons enable ingress
    ▪ Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
    ▪ Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.1.1
    ▪ Using image k8s.gcr.io/ingress-nginx/controller:v1.1.0
🔎  Verifying ingress addon...
🌟  The 'ingress' addon is enabled
```
```
❯ kubectl get ns
NAME              STATUS   AGE
default           Active   39s
ingress-nginx     Active   25s
kube-node-lease   Active   41s
kube-public       Active   41s
kube-system       Active   41s
````

### CODE REVIEW

File: deploy/addons/ingress/ingress-deploy.yaml.tmpl [hunk 1]
```diff
@@ -299,7 +299,7 @@ metadata:
   namespace: ingress-nginx
 spec:
   type: NodePort
-  {{- if eq .IngressAPIVersion "v1"}}
+  {{- if and (eq .IngressAPIVersion "v1") (not .LegacyKubernetesVersion)}}
```
Comment (medyagh) [inline deploy/addons/ingress/ingress-deploy.yaml.tmpl:302]: I think we could choose a better future-proof , since this field is used by all other addons,

Comment (medyagh) [reply deploy/addons/ingress/ingress-deploy.yaml.tmpl:302]: I would prefer any other name than legacyKubernetesVersion.
even something as preOneTwentyKuberentes

Comment (prezha) [reply deploy/addons/ingress/ingress-deploy.yaml.tmpl]: it makes sense (and i like it!) - renamed accordingly

**Comment by medyagh** [comment]
Thanks @prezha looks like u might need to redo the CLA https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/18706487/56353740/13173/#/?version=2

**Comment by prezha** [comment]
> Thanks @prezha looks like u might need to redo the CLA https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/18706487/56353740/13173/#/?version=2

just did (2nd time!)

**Review by sharifelgamal — APPROVED** [review]

**Comment by dytyniuk** [comment]
Thanks @prezha 
Confirmed working with K8S v1.19.6 and the most recent one 1.22.4 🙇

**Comment by prezha** [comment]
> Thanks @prezha Confirmed working with K8S v1.19.6 and the most recent one 1.22.4 bow

excellent!
thank you, @dytyniuk for taking the time to test both old(er) and the latest versions and confirming it working!

**Pull request merged**